### ISSUE TITLE
Missing bracket  ：Count(CASE WHEN IN)

### ISSUE THREAD

**Issue by kfdjsakfdjk** [JSQLParser/JSqlParser#1239]
sql: 
SELECT COUNT(DISTINCT CASE WHEN room IN (11167, 12074, 4484, 4483, 6314, 11168, 10336, 16445, 13176, 13177, 13178) THEN uid END) AS uidCount from tableName

after parser（sql = buffer.toString()）:
SELECT COUNT(DISTINCT CASE WHEN room IN 11167, 12074, 4484, 4483, 6314, 11168, 10336, 16445, 13176, 13177, 13178 THEN uid END) AS uidCount

issue： Missing bracket

**Comment by manticore-projects** [comment]
Confirmed with 4.1-Snapshot latest GIT
```java
@Test
    public void testMissingBracketsNestedInIssue() throws JSQLParserException {
        assertSqlCanBeParsedAndDeparsed("SELECT COUNT(DISTINCT CASE WHEN room IN (11167, 12074, 4484, 4483, 6314, 11168, 10336, 16445, 13176, 13177, 13178) THEN uid END) AS uidCount from tableName", true);
    }
```
--> fails

JSQLFormatter deparses it correctly though. [You can test it here online](http://jsqlformatter.manticore-projects.com/jsqlformatter/demo.html?args=-c%20MoUQMiDCAqAEkHsCuA7ALgClgEQJLGlwDkZ4BBUAKFhtrvtgHUAJEI2AJwQQFtZjYWWAEZRANgDsAGhEAmAAwSALDKVKAHEuoMduvbFUaAzDLFHhKkePXb9dvTOHyjRsY7FqArI-MSxt+0C6H2EJaRFfdVgASgCgmmhWdiQASwATOJ02bBjYClhUtIBjZHRKADEAJQB5AFlYNABDACMAGwBTFEaedsoAbiA+]

I will look into this.

**Comment by kfdjsakfdjk** [comment]
When will version 4.1 be released

**Comment by manticore-projects** [comment]
Thank you for the Test Cases, it was very helpful and I have added it to the Test Suite.
Solved with PR #1240.

@wumpz: Please fast track this because a few things were really broken and we did not have Test Cases to catch it yet.

**Comment by wumpz** [comment]
merged
